### PR TITLE
UX-614 Convert Button to typescript

### PR DESCRIPTION
**Pull request by jonambas**
Requires #907 to be merged first

### What Changed
- Converts `Button` to typescript
- Add a type helper for polymorphic components with `as`

### How To Test or Verify
- Verify libby button stories work
- Verify Button tests pass

### PR Checklist

- [ ] Add the correct `type` label
- [ ] Pull request approval from #uxfe or #design-guild

### CODE REVIEW

File: packages/matchbox/src/components/Button/Button.tsx [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+      onFocus,
+      onBlur,
+      buttonSize,
+      buttonVariant,
+      buttonColor,
+      ref,
+      loading,
+      ...systemProps,
+      ...componentProps,
+    };
```
Comment (logansparlin) [inline packages/matchbox/src/components/Button/Button.tsx:242]: Should we add types for these?

File: packages/matchbox/src/components/Button/Group.tsx [hunk 2]
```diff
@@ -4,11 +4,16 @@ import styled from 'styled-components';
...
+  className?: string;
+  children?: React.ReactNode;
+};
+
 const StyledGroup = styled(Box)`
   ${group}
 `;
 
-const Group = ({ children, className = '' }) => (
+const Group: React.FC<GroupProps> = ({ children, className = '' }) => (
```
Comment (logansparlin) [inline packages/matchbox/src/components/Button/Group.tsx:16]: I don't think we need React.FC here since you're explicitly typing className and children

**Review by logansparlin — APPROVED** [review]
Looks good, just a couple small comments

File: packages/matchbox/src/components/Button/Button.tsx (hunk 1 above)
Comment (jonambas) [reply packages/matchbox/src/components/Button/Button.tsx:242]: These are already typed no?

Comment (logansparlin) [reply packages/matchbox/src/components/Button/Button.tsx:242]: oh yea my bad, not sure what I was thinking 😆

**Pull request merged**